Device.characteristics() is empty

This is an example from the `README`:  
```rust
light.discover_characteristics().unwrap();

// find the characteristic we want
let chars = light.characteristics();
let cmd_char = chars.iter().find(|c| c.uuid == UUID::B16(0xFFE9)).unwrap();
```

In my case, `light.characteristics()` is empty (at least on Windows).  

It took me some time to figure out that characteristics can be obtained directly from the `dicsover_characteristics()` method:  
```rust
let cmd_char = light.discover_characteristics().unwrap().iter().find(|c| c.uuid == UUID::B16(0xFFE9)).unwrap();;
```

In my case, I'm connecting to the IKEA desk instead of Bluetooth light, but it shouldn't make any difference here.  

I didn't test that on Linux.

@qwandor - I'm not sure if I'm doing something wrong, but this branch doesn't work for me. I took the example code from the docs, with slight modifications:

```rust
use std::thread;
use std::time::Duration;
#[cfg(target_os = "linux")]
use btleplug::bluez::{adapter::Adapter, manager::Manager};
#[cfg(target_os = "windows")]
use btleplug::winrtble::{adapter::Adapter, manager::Manager};
#[cfg(target_os = "macos")]
use btleplug::corebluetooth::{adapter::Adapter, manager::Manager};
use btleplug::api::{Central, Peripheral, WriteType};

const DEVICE_NAME: &str = "Desk";

pub fn main() {
    let manager = Manager::new().unwrap();

    let adapters = manager.adapters().unwrap();
    let central = adapters.into_iter().nth(0).unwrap();

    central.start_scan().unwrap();
    thread::sleep(Duration::from_secs(2));

    let device = central.peripherals().into_iter()
        .find(|p| p.properties().local_name.iter()
            .any(|name| name.contains(DEVICE_NAME))).unwrap();

    device.connect().unwrap();

    println!(
        "discover_characteristics(): {:?}",
        device.discover_characteristics()
    );
    println!("characteristics(): {:?}", device.characteristics());
}

```

If I'll use code from the master, everything works as described here. Dependencies:  
```toml
[dependencies]
btleplug = { git="https://github.com/deviceplug/btleplug.git" }
```

Output:  
```
discover_characteristics(): Ok([Characteristic { start_handle: 0, ...
characteristics(): {}
```

Now, when I change dependencies to the one below, the code doesn't compile:
```toml
[dependencies]
btleplug = { git="https://github.com/deviceplug/btleplug.git", branch="characteristics" }
```

```rust
error[E0603]: module `winrtble` is private
   --> src\main.rs:12:15
    |
12  | use btleplug::winrtble::manager::Manager;
    |               ^^^^^^^^ private module
    |
note: the module `winrtble` is defined here
   --> C:\Users\aklajnert\.cargo\git\checkouts\btleplug-4193de4d1cdc37dd\1811316\src/lib.rs:105:1
    |
105 | mod winrtble;
    | ^^^^^^^^^^^^^
```

I don't understand why the module is now private, in the PR I can't see any change that would do that...

--- a/src/winrtble/ble/characteristic.rs
+++ b/src/winrtble/ble/characteristic.rs
@@ -12,7 +12,11 @@
 // Copyright (c) 2014 The Rust Project Developers
 
 use super::super::bindings;
-use crate::{api::WriteType, Error, Result};
+use crate::{
+    api::{Characteristic, WriteType},
+    winrtble::utils,
+    Error, Result,
+};
 
 use bindings::windows::devices::bluetooth::generic_attribute_profile::{
     GattCharacteristic, GattClientCharacteristicConfigurationDescriptorValue,
@@ -33,6 +37,7 @@ impl Into<GattWriteOption> for WriteType {
     }
 }
 
+#[derive(Debug)]
 pub struct BLECharacteristic {
     characteristic: GattCharacteristic,
     notify_token: Option<EventRegistrationToken>,
@@ -124,6 +129,13 @@ impl BLECharacteristic {
             .unwrap();
         trace!("unsubscribe {:?}", status);
         Ok(())
+    }
+
+    pub fn to_characteristic(&self) -> Characteristic {
+        let uuid = utils::to_uuid(&self.characteristic.uuid().unwrap());
+        let properties =
+            utils::to_char_props(&self.characteristic.characteristic_properties().unwrap());
+        Characteristic { uuid, properties }
     }
 }
 

--- a/src/winrtble/peripheral.rs
+++ b/src/winrtble/peripheral.rs
@@ -46,7 +46,6 @@ pub struct Peripheral {
     adapter: AdapterManager<Self>,
     address: BDAddr,
     properties: Arc<Mutex<PeripheralProperties>>,
-    characteristics: Arc<Mutex<BTreeSet<Characteristic>>>,
     connected: Arc<AtomicBool>,
     ble_characteristics: Arc<DashMap<Uuid, BLECharacteristic>>,
     notification_senders: Arc<Mutex<Vec<UnboundedSender<ValueNotification>>>>,
@@ -58,7 +57,6 @@ impl Peripheral {
         let mut properties = PeripheralProperties::default();
         properties.address = address;
         let properties = Arc::new(Mutex::new(properties));
-        let characteristics = Arc::new(Mutex::new(BTreeSet::new()));
         let connected = Arc::new(AtomicBool::new(false));
         let ble_characteristics = Arc::new(DashMap::new());
         let notification_senders = Arc::new(Mutex::new(Vec::new()));
@@ -67,7 +65,6 @@ impl Peripheral {
             adapter,
             address,
             properties,
-            characteristics,
             connected,
             ble_characteristics,
             notification_senders,
@@ -194,11 +191,10 @@ impl Debug for Peripheral {
             ""
         };
         let properties = self.properties.lock().unwrap();
-        let characteristics = self.characteristics.lock().unwrap();
         write!(
             f,
             "{} properties: {:?}, characteristics: {:?} {}",
-            self.address, *properties, *characteristics, connected
+            self.address, *properties, self.ble_characteristics, connected
         )
     }
 }
@@ -220,8 +216,10 @@ impl ApiPeripheral for Peripheral {
     /// The set of characteristics we've discovered for this device. This will be empty until
     /// `discover_characteristics` is called.
     fn characteristics(&self) -> BTreeSet<Characteristic> {
-        let l = self.characteristics.lock().unwrap();
-        l.clone()
+        self.ble_characteristics
+            .iter()
+            .map(|item| item.value().to_characteristic())
+            .collect()
     }
 
     /// Returns true iff we are currently connected to the device.
@@ -270,15 +268,13 @@ impl ApiPeripheral for Peripheral {
         if let Some(ref device) = *device {
             let mut characteristics_result = vec![];
             let characteristics = device.discover_characteristics().await?;
-            for characteristic in characteristics {
-                let uuid = utils::to_uuid(&characteristic.uuid().unwrap());
-                let properties =
-                    utils::to_char_props(&characteristic.characteristic_properties().unwrap());
-                let chara = Characteristic { uuid, properties };
-                characteristics_result.push(chara);
+            for gatt_characteristic in characteristics {
+                let ble_characteristic = BLECharacteristic::new(gatt_characteristic);
+                let characteristic = ble_characteristic.to_characteristic();
                 self.ble_characteristics
-                    .entry(uuid)
-                    .or_insert_with(|| BLECharacteristic::new(characteristic));
+                    .entry(characteristic.uuid.clone())
+                    .or_insert_with(|| ble_characteristic);
+                characteristics_result.push(characteristic);
             }
             return Ok(characteristics_result);
         }

